src/workspace.rs

```diff
@@ -0,0 +1,23 @@
+use super::error::{MainError, PnError};
+use lets_find_up::*;
+use std::path::{Path, PathBuf};
+
+const WORKSPACE_MANIFEST_FILENAME: &str = "pnpm-workspace.yaml";
+
+pub fn find_workspace_root(cwd: &dyn AsRef<Path>) -> Result<PathBuf, MainError> {
+    let workspace_manifest_location = find_up_with(
+        WORKSPACE_MANIFEST_FILENAME,
+        FindUpOptions {
+            kind: FindUpKind::File,
+            cwd: cwd.as_ref(),
+        },
+    )
+    .map_err(MainError::from_dyn)?;
+    match workspace_manifest_location {
+        Some(path) => Ok(path
+            .parent()
+            .ok_or(MainError::Pn(PnError::NotInWorkspace))?
```

Comment:
I am not sure about this line. I don't think this is the right error but I also don't think this error will ever happen.